app/src/components/widgets/charts/launchesDurationChart/launchesDurationChart.jsx

```diff
@@ -0,0 +1,197 @@
+    this.getConfig();
+  }
+
+  componentWillUnmount() {
+    this.node.removeEventListener('mousemove', this.getCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeChart);
+  }
+
+  onChartCreated = (chart, element) => {
+    this.chart = chart;
+    this.node = element;
+
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    this.resizeChart();
+
+    this.node.addEventListener('mousemove', this.getCoords);
+  };
+
+  getPosition = (d, width, height) => {
+    const rect = this.node.getBoundingClientRect();
+    const left = this.x - rect.left - width / 2;
+    const top = this.y - rect.top - height;
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getConfig = () => {
+    const { widget, isPreview, container } = this.props;
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+
+    const { timeType, chartData, itemData } = prepareChartData(widget);
+    this.itemData = itemData || [];
+    this.timeType = timeType;
+
+    this.config = {
+      data: {
+        columns: [chartData],
+        type: 'bar',
+        colors: {
+          [DURATION]: COLOR_CHART_DURATION,
+        },
+        groups: [[DURATION]],
+        color: (color, d) => {
+          if (itemData[d.index] && isValueInterrupted(itemData[d.index])) {
+            return COLOR_FAILED;
+          }
+          return color;
+        },
+      },
+      grid: {
+        y: {
+          show: !isPreview,
+        },
+      },
+      axis: {
+        rotated: true,
+        x: {
+          show: !isPreview,
+          type: 'category',
+          categories: itemData.map(transformCategoryLabel),
+          tick: {
+            values: getLaunchAxisTicks(itemData.length),
+            width: 60,
+            centered: true,
+            inner: true,
+            multiline: false,
+            outer: false,
+          },
+        },
+        y: {
+          show: !isPreview,
+          tick: {
+            format: (d) => (parseInt(d, 10) / timeType.value).toFixed(2),
+          },
+          padding: {
+            top: 0,
+            bottom: 0,
+          },
+          label: {
+            text: 'seconds',
+            position: 'outer-center',
+          },
+        },
+      },
+      interaction: {
+        enabled: !isPreview,
+      },
+      padding: {
+        top: isPreview ? 0 : 20,
+        left: isPreview ? 0 : 40,
+        right: isPreview ? 0 : 20,
+        bottom: isPreview ? 0 : 10,
+      },
+      legend: {
+        show: false,
+      },
+      tooltip: {
+        grouped: true,
+        position: this.getPosition,
+        contents: this.renderContents,
+      },
+      size: {
+        height: this.height,
+      },
+    };
+
+    this.setState({
+      isConfigReady: true,
+    });
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  resizeChart = () => {
```

Comment:
Can it be handled by the `C3Chart` component itself?

Reply:
@DiscoElevator The resize algorithm is the same as it is in the reference Chart component (`LaunchesComparisonChart`). It reacts to the wrapping components size changing, and resizes the C3Chart explicitly. I believe this solution should be kept unless we see a recurring pattern in the handling of resize events.